library/asn1write.c

```diff
@@ -82,8 +82,10 @@ int mbedtls_asn1_write_len( unsigned char **p, unsigned char *start, size_t len
         *--(*p) = 0x83;
         return( 4 );
     }
-
+	
+#if ((UINT_MAX) != 0xffffffffu)//relevant only for 64 bit platforms. on 32 bit platforms, size_t should always be 32 bits
```

Comment:
I do not think the preprocessor directive `#if ((UINT_MAX) != 0xffffffffu)` can reliably tell you that the architecture is either 32 or 64 bits. The documentation for `UINT_MAX` says: `Maximum value for an object of type unsigned int (65535 (216-1) or greater)`. In other words, depending on the library implementation you might get a 16, 32, 64 bit type or anything else. For example, in my ubuntu 64-bit machine I did the following experiment:
```C
printf("sizeof(int):%zu, sizeof(unsigned int):%zu, sizeof(size_t):%zu\n",
    sizeof(int), sizeof(unsigned int), sizeof(size_t));
printf("UINT_MAX:%u, UINT_MAX==unsigned int:%d\n",
    UINT_MAX, UINT_MAX == ((unsigned int)-1));
```
And this prints:
```
sizeof(int):4, sizeof(unsigned int):4, sizeof(size_t):8
UINT_MAX:4294967295, UINT_MAX==unsigned int:1
```
Even though my machine is 64-bit, the sizeof(unsigned int) is still a 32-bit value.